frontend/simple/views/containers/Proposals.vue

```diff
@@ -55,138 +61,130 @@
+          ownVote: proposals[hash].for.includes(userData.name) || proposals[hash].against.includes(userData.name)
+            ? proposals[hash].for.includes(userData.name)
+            : null,
+          isOwnProposal: proposals[hash].initiator === userData.name,
+          initiator: proposals[hash].initiator,
+          hash: hash
+        }
+
+        if (proposalData.isOwnProposal) {
+          own.push(proposalData)
+        } else if (proposalData.ownVote === null) {
+          notVoted.push(proposalData)
+        } else if (proposalData.ownVote !== null) {
+          alreadyVoted.push(proposalData)
         }
       }
 
       return { notVoted, alreadyVoted, own }
     },
     allVoted () {
-      const { notVoted, alreadyVoted } = this.groupProposals
+      const { notVoted, alreadyVoted } = this.proposalList
 
       return alreadyVoted.length && notVoted.length === 0
     }
   },
   methods: {
-    handleVoteAgainst () {
-      console.log('TODO Logic - The user voted against')
+    async handleVoteAgainst (hash) {
+      try {
+        // Create a vote against the proposal
+        const groupId = this.$store.state.currentGroupId
+        const latest = await backend.latestHash(groupId)
+        const vote = new Events.HashableGroupVoteAgainstProposal({
+          username: this.currentUserIdentityContract.attributes.name,
+          proposalHash: hash
+        }, latest)
+        await backend.publishLogEntry(groupId, vote)
+      } catch (ex) {
+        // TODO: diplay error to user
+        console.error('Failed to cast vote:')
+        console.log(ex)
+      }
     },
-    handleVoteFor () {
-      console.log('TODO Logic - The user voted for')
+    async handleVoteFor (hash) {
+      try {
+        // Create a vote for the proposal
+        const groupId = this.$store.state.currentGroupId
+        let latest = await backend.latestHash(groupId)
+        const vote = new Events.HashableGroupVoteForProposal({
+          username: this.currentUserIdentityContract.attributes.name,
+          proposalHash: hash
+        }, latest)
+        await backend.publishLogEntry(groupId, vote)
+
+        // If the vote passes fulfill the action
+        const proposal = this.$store.getters.proposalData(hash)
+        const memberCount = Object.entries(this.currentGroupState.profiles).length
+        const threshold = Math.ceil(proposal.threshold * memberCount)
+        if (proposal.for.length + 1 >= threshold) {
+          let lastActionHash = null
+          const actionDate = new Date().toString()
+          for (let step of proposal.actions) {
+            latest = await backend.latestHash(step.contractId)
+            let actObj = JSON.parse(template(step.action, {lastActionHash, actionDate}))
+            let entry = new Events[actObj.type](actObj.data, latest)
+            lastActionHash = entry.toHash()
+            await backend.publishLogEntry(step.contractId, entry)
+          }
```

Comment:
Is this block of code to handle when the proposal is approved ? Shouldn't it be in a separate method?

Reply:
could be, I tried to leave an explaining comment but I'll move it out of here :)